### PR TITLE
fix: delay write cookie headers to first write

### PR DESCRIPTION
**Pull request by PatrickJS**
# Overview
fixes https://github.com/BuilderIO/qwik/issues/5951

# What is it?

- [ ] Feature / enhancement
- [x] Bug
- [ ] Docs / tests / types / typos

# Description

set headers once but allow for dev to add cookie values

### CODE REVIEW

File: packages/qwik-city/middleware/node/http.ts [hunk 1]
```diff
@@ -89,13 +89,18 @@ export async function fromNodeHttp(
     },
     getWritableStream: (status, headers, cookies) => {
       res.statusCode = status;
-      headers.forEach((value, key) => res.setHeader(key, value));
-      const cookieHeaders = cookies.headers();
-      if (cookieHeaders.length > 0) {
-        res.setHeader('Set-Cookie', cookieHeaders);
-      }
+      let once = false;
```
Comment (gioboa) [inline packages/qwik-city/middleware/node/http.ts:92]: This is an oversight I guess

**Review by gioboa — CHANGES_REQUESTED** [review]

File: packages/qwik-city/middleware/node/http.ts (hunk 1 above)
Comment (PatrickJS) [reply packages/qwik-city/middleware/node/http.ts:92]: I want to write once since write can happen multiple times (unless it's renderToString)

Comment (gioboa) [reply packages/qwik-city/middleware/node/http.ts:92]: I see. I think this should be true to execute the next if once

Comment (PatrickJS) [reply packages/qwik-city/middleware/node/http.ts:92]: oh yeah I meant to set the other one to true

**Review by gioboa — APPROVED** [review]

**Comment by PatrickJS** [comment]
@wmertens we need to delay writing to headers until we first "write" to the client. that way any app code can run and add cookie headers before we stream.

**Comment by PatrickJS** [comment]
yes this applies to all headers. the e2e tests hang with this fix but when I run it locally it works correctly in preview-only. so there is something going on. @wmertens is there a better way to hook into this?

**Comment by wmertens** [comment]
I don't think there's a better way because each adapter had their own way of sending cookies.

I think the e2e problem may be when there's no write and the response just closes without sending the cookies? Odd but maybe.

In any case you should handle `once` still being true on close.

**Comment by wmertens** [comment]
@PatrickJS notice that the e2e tests fail on this

**Review by wmertens — APPROVED** [review]
Looks good!

**Comment by wmertens** [comment]
...but sadly the e2e fails. Try removing the once=false in close? Not sure that's needed or even desired

**Comment by PatrickJS** [comment]
@wmertens I'm thinking e2e is waiting for headers which is why it fails during e2e but not when running locally. maybe there's another way

**Comment by PatrickJS** [comment]
I think it's better having devs use `plugin@<name>.ts` and middleware for this

**Pull request closed without merging**